src/processing/yarnrc.ts

```diff
@@ -2,17 +2,74 @@ import { assert, isPlainObject } from '@metamask/utils';
 import execa from 'execa';
 import { readFile, rm, writeFile } from 'fs/promises';
 import { dump, load } from 'js-yaml';
-import { Ora } from 'ora';
 import { resolve } from 'path';
 import semver from 'semver';
 
 import { TEMPORARY_PATH } from './files';
+import { TaskOptions } from '../options';
 import { pathExists, warn } from '../utils';
 
 const LEGACY_YARN_PATH = resolve(process.cwd(), '.yarnrc');
 const CURRENT_YARN_PATH = resolve(process.cwd(), '.yarnrc.yml');
 const TEMPLATE_YARN_PATH = resolve(TEMPORARY_PATH, '.yarnrc.yml');
 
+/**
+ * Check for a legacy `.yarnrc` file and delete it if it exists. If the check
+ * option is enabled, a warning will be displayed instead.
+ *
+ * @param options - The options for the task.
+ * @param options.spinner - The spinner to use for logging.
+ * @param options.check - Whether to only check for changes compared to the
+ * template. When this is enabled, no files will be modified.
+ */
+async function checkLegacyYarnRc({ spinner, check }: TaskOptions) {
+  const legacyYarnRcExists = await pathExists(LEGACY_YARN_PATH);
+  if (!legacyYarnRcExists) {
+    return;
+  }
+
+  if (check) {
+    warn(
+      spinner,
+      'Legacy .yarnrc file exists. It will not be deleted because the --check flag is enabled.',
+    );
```

Comment:
Missing return, meaning this message is a lie 😅

Reply:
Oops. 😅